core-api/src/main/java/com/optimizely/ab/config/audience/match/MatchType.java

```diff
@@ -64,9 +65,11 @@ public static MatchType getMatchType(String matchType, Object conditionValue) {
                 }
                 break;
             default:
+                logger.warn(String.format("Audience condition \"%s\" uses an unknown match type: %s", userAttribute.toString(), matchType));
                 return new MatchType(matchType, new NullMatch());
         }
 
+        logger.warn(String.format("Audience condition \"%s\" condition value \"%s\" data type is inapplicable", userAttribute.toString(), conditionValue));
```

Comment:
not justified as `warn` level IMO.

this would be more informative if we logged `matchType`.
also, think we shouldn't be logging `conditionValue` object -- it is potentially very expensive (depends on it's `toString()` impl) and has potential to log undesired data (PII).

Reply:
> not justified as `warn` level IMO.

I dunno, a customer probably won't expect an audience condition to become inactive for this reason, and therefore probably ought to be warned. If it helps, the updated log message suggests that the customer upgrade their SDK.

> this would be more informative if we logged `matchType`.

It's included (along with `type`, `name`, and `value`) in our logging of the audience condition (the `userAttribute` variable).

> also, think we shouldn't be logging `conditionValue` object -- it is potentially very expensive (depends on it's `toString()` impl) and has potential to log undesired data (PII).

Good feedback. We're not logging it any more!